### PR TITLE
added dark mode for analog clock

### LINKED ISSUE #976: Dark mode for Analog Clock
Hey there, I have created a dark mode toggle for the analog clock. Check it OUT.
ASTRO Javascript Clock UI Dark - Google Chrome 2022-10-14 16-54-10

### PR DESCRIPTION
**Pull request by aysal04**
### 🛠️ Fixes #976 

Closes #976  

### 👨‍💻 Changes proposed

I have added a toggle checkbox that allows us to switch between light and darkmodes. For this I have used basic java script to change the colors based on the status of the checkbox.

### ✅ Check List (Check all the applicable boxes) 

- [x] You have linked this PR to the correct issue.
- [x] My code doesn't break any part of the project (Zero Octave-Javascript-Projects).
- [x] This PR does not contain plagiarized content.
- [x] My Addition/Changes works properly and matches the overall repo pattern.
- [x] The title of my pull request is a short description of the requested changes.

### 📄 Note to reviewers

As the color of the numbers on the clock is fixed, we couldn't change them. However grey suits both the themes. Here is the preview of light and dark mode UI/UX designs of Analog clock.

Analog Clock UI Light
Analog Clock UI Dark

### CODE REVIEW

File: assets/css/Analog_clock.css
Comment (Anmol-Baranwal) [inline assets/css/Analog_clock.css:139]:
```suggestion
  top: 3px;
```
for better alignment

File: assets/css/Analog_clock.css
Comment (Anmol-Baranwal) [inline assets/css/Analog_clock.css:136]:
```suggestion
  background-color: black;
```
Making default color `black` differentiates the ball from background

File: assets/Js/analog_clock1.js
Comment (Anmol-Baranwal) [inline assets/Js/analog_clock1.js:32]:
```suggestion
const checkbox = document.getElementById("checkbox");
const body = document.querySelector("body");
const label = document.querySelector(".label");
const root = document.querySelector(":root");
const ball = document.querySelector(".ball");

checkbox.addEventListener("change", function () {
  if (this.checked) {
    body.style.background = "black";
    label.style.backgroundColor = "#40DCA5";
    root.style.setProperty("--sec-color", "#fff");
    ball.style.backgroundColor = "#fff"; // Change ball color to white
  } else {
    body.style.background = "skyblue";
    label.style.backgroundColor = "white";
    root.style.setProperty("--sec-color", "red");
    ball.style.backgroundColor = "black"; // Change ball color to black
  }
});
```

This will change the color on toggle of checkbox, which is good for accessibility and contrast

File: Public/Analog_clock.html
Comment (Anmol-Baranwal) [inline Public/Analog_clock.html:70]: add this to link js file
```javscript
    <script src="../assets/Js/analog_clock1.js"></script>
```

just before ending of body tag

File: Public/Analog_clock.html
Comment (Anmol-Baranwal) [inline Public/Analog_clock.html:67]: It is recommended to remove the JavaScript code from the HTML file since it doesn't make much sense to have it there when there is already an external JavaScript file attached. By separating the JavaScript code into its own file, it enhances code organization and maintainability.

**Review by Anmol-Baranwal — CHANGES_REQUESTED** [review]
Also remove this JavaScript code from html file
```javascript
<script type="text/javascript">
  const deg = 6;
  const hr = document.querySelector("#hr");
  const mn = document.querySelector("#mn");
  const sc = document.querySelector("#sc");

  setInterval(() => {
    let day = new Date();
    let hh = day.getHours() * 30;
    let mm = day.getMinutes() * deg;
    let ss = day.getSeconds() * deg;

    hr.style.transform = `rotateZ(${hh + mm / 12}deg)`;
    mn.style.transform = `rotateZ(${mm}deg)`;
    sc.style.transform = `rotateZ(${ss}deg)`;
  });
</script>
```

@aysal04 
Please make the suggested changes to the code. Once done, the code will be ready for merging. Best of luck, and feel free to ask if you have any doubts or questions about the code.

**Review by Anmol-Baranwal — CHANGES_REQUESTED** [review]
Hi @aysal04,

just a minor change:

kindly add this code in html just above script external js file code

```javascript
<script
      data-name="BMC-Widget"
      data-cfasync="false"
      src="https://cdnjs.buymeacoffee.com/1.0.0/widget.prod.min.js"
      data-id="Astrodevil"
      data-description="Support me on Buy me a coffee!!"
      data-message=""
      data-color="#40DCA5"
      data-position="Right"
      data-x_margin="50"
      data-y_margin="50"
    ></script>
```

I think you removed this by mistake.

You can do this tomorrow as well, I will merge the PR after this.

**Comment by aysal04** [comment]
Yeah sorry. I deleted it accidentally. Currently I have made some changes to address issue #928 . Can I commit both the changes together here? Or shall a open a new PR?

**Comment by Anmol-Baranwal** [comment]
> Yeah sorry. I deleted it accidentally. Currently I have made some changes to address issue #928 . Can I commit both the changes together here? Or shall a open a new PR?

It is always recommended to work on different issues using separate branches in order to maintain a clean history. 

Additionally, since the issue in question has not been assigned to you, you can proceed with working on it if the assigned person does not submit a pull request (PR) in a timely manner.

Just commit the changes for this pull request here and refrain from addressing any other issues in this pull request.

**Review by Anmol-Baranwal — APPROVED** [review]
Appreciate your contribution :)

**Comment by aysal04** [comment]
> Thank you @aysal04 for taking out your valuable time in order to contribute to our project. Looking forward for more such amazing contributions :)

It was an amazing experience contributing for the first time to an open source. Thanks to my mentor for such an amazing guidance

**Comment by aysal04** [comment]
> It was really an amazing experience contributing for the first time to an open source. Thanks to you for being so supportive.

**Comment by aysal04** [comment]
> Appreciate your contribution :)

It was really an amazing experience contributing for the first time to an open source. Thanks to you for being so supportive

**Pull request merged**
